doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,297 @@ Reasoning with inductive types
+      Provides explicit instances for
+      the :term:`dependent premises <dependent premise>` of the type of
+      :token:`one_term`.
+
+   :n:`@occurrences`
+     Selects specific subterms of the goal and/or hypotheses to apply
+     the tactic to.  See :ref:`Occurrence clauses <occurrenceclauses>`.
+     If it occurs at the end of the :n:`@induction_clause_list`, then
+     there can only be one :n:`induction_clause`, which can't have its
+     own :n:`occurrences` clause.
+
+   .. tacn:: edestruct @induction_clause_list
+
+      Equivalent to :n:`destruct @term` except that it does not
+      fail if the instance of a dependent premises of the type
+      of :token:`term` is not inferrable. Instead, the unresolved instances
+      are left as existential variables to be inferred later, in the same way
+      they are in :tacn:`eapply`.
 
-   The tactic :n:`case` is a more basic tactic to perform case analysis without
-   recursion. It behaves as :n:`elim @term` but using a case-analysis
-   elimination principle and not a recursive one.
+.. tacn:: case @induction_clause_list
 
-.. tacv:: case @term with @bindings
+   .. todo PR: clear as mud? (below) the syntax accepts more parameters than elim
+      not sure what to say here
 
-   Analogous to :n:`elim @term with @bindings` above.
+   An older, more basic tactic to perform case analysis without
+   recursion. It behaves like :tacn:`elim` but uses a case-analysis
+   elimination principle rather than a recursive one.
+   `case` only modifies the goal; it does not modify the :term:`local context`.
 
-.. tacv:: ecase @term {? with @bindings }
-   :name: ecase
+.. tacn:: ecase @induction_clause_list
 
-   In case the type of :n:`@term` has dependent premises, or dependent premises
-   whose values are not inferable from the :n:`with @bindings` clause,
+   If the type of :n:`@one_term` (in :n:`@destruction_arg`) has
+   :term:`dependent premises <dependent premise>`, or dependent premises
+   whose values are not inferrable from the :n:`with @bindings` clause,
    :n:`ecase` turns them into existential variables to be resolved later on.
 
-.. tacv:: simple destruct @ident
-   :name: simple destruct
+.. tacn:: casetype @one_term
+   :undocumented:
 
-   This tactic behaves as :n:`intros until @ident; case @ident` when :n:`@ident`
-   is a quantified variable of the goal.
+.. tacn:: simple destruct {| @ident | @natural }
 
-.. tacv:: simple destruct @natural
+   Equivalent to :tacn:`intros` :n:`until {| @ident | @natural }; case @ident`
+   where :n:`@ident` is a quantified variable of the goal and otherwise fails.
 
-   This tactic behaves as :n:`intros until @natural; case @ident` where :n:`@ident`
-   is the name given by :n:`intros until @natural` to the :n:`@natural` -th
-   non-dependent premise of the goal.
+.. tacn:: dependent destruction @ident {? generalizing {+ @ident } } {? using @one_term }
+   :undocumented:
 
-.. tacv:: case_eq @term
+.. tacn:: case_eq @one_term
```

Comment:
You can add to the description of this tactic that we recommend using `destruct` with the `eqn` clause instead. Maybe you can also move it to make it appear as a variant of `case`.